### DISCUSSION TITLE
Does this facilitate scheduling DB migrations?

### DISCUSSION THREAD

**Discussion by dhnaranjo** [Shopify/maintenance_tasks#710] [Q&A] [answered]
Hello!

I'm wondering if I can use this tool to schedule DB modifications. It'd be rad if I could do things like:

- Add a column in a db migration
- Populate it in a maintenance task
- Make that column disallow null values

I suspect that the answer is no, but I haven't found a clear answer anywhere round the repo.

**Comment by etiennebarrie** [comment, marked as answer]
Yes it's a bit surprising we don't even have a sentence in the README to explain that, but that's precisely one of the core use cases we wanted to tackle.
I'm not sure about your wording about scheduling database migrations, to be clear the gem only helps with the step 2 here. Step 1 and 3 are regular migrations (or online schema changes tools like LHM).

**Reply by dhnaranjo** [reply]
By schedule I meant being able to sequence those operations without multiple deploys. That'd be the dream, but this is still a rad tool for the middle but. Thanks.

**Reply by etiennebarrie** [reply]
That depends on how you run your migrations. If they don't run automatically, you could ship one PR with the 2 migrations and the maintenance tasks, and then run the first migration, the task, the second migration. But yeah if migrations run automatically at deploy time, then you'll need at least two PRs.
